config/environment_test.go

```diff
@@ -102,6 +102,22 @@ func TestRemoveEnvOverrides(t *testing.T) {
 				in.SqlSettings.DataSourceReplicas = []string{"otherthing", "alsothis"}
 			}),
 		},
+		{
+			name: "map[string]map[string]interface{}",
+			inputConfig: modifiedDefault(func(in *model.Config) {
+				in.PluginSettings.Plugins = make(map[string]map[string]interface{})
+				in.PluginSettings.Plugins["test"] = make(map[string]interface{})
+				in.PluginSettings.Plugins["test"]["test2"] = ""
+			}),
+			env: map[string]string{
+				"MM_PLUGINSETTINGS_PLUGINS_TEST_TEST2": "some_value",
```

Comment:
Can we also have a test for `MM_PLUGINSETTINGS_PLUGINSTATES_TEST_ENABLE` = `true`

Reply:
There's no way to understand underlying type for interface in [environment.go#173](https://github.com/mattermost/mattermost-server/pull/20237/files#diff-f8292d4dd253d4b3b7c8c6e5e85860e3c43a43cc2d76feb659ca4a6b6b236d86R173) 
So this is basically restrictions, that only string values may be set by this method. 

If you know the correct way to get underlying type, i would like to implement this. Otherwise it is like it is.

Seems to work with first level strings only. Better then nothing

Reply:
If we see the string's value is `true` or `false`, are we able to convert this into a `bool` before setting the value?